### PR TITLE
[Feature] - New <SchoolCard> component

### PR DESCRIPTION
**Pull request by kylemh**
# Description of changes

Implements re-imagines School Card.

# On Hold
I don't like this code. I need to look at it all from a higher level to try to obfuscate better. I didn't enjoy re-making a component to be able to map it within Storybook - that's flimsy if the component itself ever changes. Opening this PR for review and to get ideas. I'll re-approach this another day.

_TODO_: 
- Edit the close icon so the lines are thicker
- Add a prop to conditionally render the circle background to the close icon in the CloseButton component
- Adjust code schools logos in s3 to work for this component.
- Give OutboundLink component the ability to style as a Button rather than nest it within a Button

# Issue Resolved

https://github.com/OperationCode/operationcode_frontend/issues/934

## Screenshots/GIFs

screen shot 2018-09-14 at 11 35 13 am
screen shot 2018-09-14 at 11 35 17 am

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 663fad15c5fe05960d3346716dffdad25440e411

https://deploy-preview-137--operation-code-storybook.netlify.com

File: components/Cards/SchoolCard/__tests__/SchoolCard.test.js [hunk 1]
```diff
@@ -5,33 +5,71 @@ import createSnapshotTest from 'test-utils/createSnapshotTest';
...
+    doesAcceptGIBill: false,
+  },
+  {
+    city: 'New York',
+    state: 'NT',
+    doesAcceptGIBill: true,
+  },
+];
+
+// const mockManyLocationsWithoutGIAccepted = [
```
Comment (togakangaroo) [inline components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: Should code commented code be removed?

Comment (kylemh) [reply components/Cards/SchoolCard/__tests__/SchoolCard.test.js]: I'm gonna use those later on for some tests to confirm business logic, such as the appearance of the text: "CITY, ST" given a single location.

**Comment by kylemh** [comment]
Nah this is gross

**Pull request closed without merging**